unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -3285,7 +3243,36 @@ bool includeInvisible
                 actionFinished(false);
             }
         }
+        
+        protected SimObjPhysics getInteractableSimObjectFromId(string objectId, bool forceVisible = false) {
+            SimObjPhysics sop = getSimObjectFromId(objectId);
+            if (sop == null) {
+                errorMessage = "Object with id '" + objectId+ "' is null";
+                return null;
+            }
+
+            if (forceVisible || IsInteractable(sop))
+            {
+                return sop;
+            } 
+
+            return null;
+        }
+
+        protected SimObjPhysics getSimObjectFromId(string objectId) {
+            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(objectId)) {
+                errorMessage = "Cannot find sim object with id '" + objectId + "'";
+                return null;
+            }
+           
+            SimObjPhysics sop = physicsSceneManager.ObjectIdToSimObjPhysics[objectId];
+            if (sop == null) {
+                errorMessage = "Object with id '" + objectId+ "' is null";
+                return null;
+            }
```

Comment:
Is this possible? Somehow I feel like this should be a critical failure of some sort as it suggests the `ObjectIdToSimObjPhysics` dictionary is broken somehow.

Reply:
Good catch. Removed.

Reply:
Good catch. Removed.